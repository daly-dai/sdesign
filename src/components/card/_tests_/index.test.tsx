import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import SCard from '../index';

import mountTest from '@dalydb/sdesign/tests/mountTest';

describe('SCard', () => {
  // 测试组件
  mountTest(SCard);
  mountTest(() => <SCard />);

  it('should render correctly', () => {
    const { container } = render(<SCard />);
    expect(container).toMatchSnapshot();
  });

  it('should support props style', () => {
    const { container } = render(<SCard style={{ backgroundColor: 'red' }} />);
    expect(container).toMatchSnapshot();
  });

  it('should support custom className', () => {
    const { container } = render(<SCard className="test-class" />);

    expect(container.querySelectorAll('.test-class').length).toBe(1);
    expect(container).toMatchSnapshot();
  });

  it('should support hasBottomPadding props', () => {
    const { container } = render(
      <SCard className="s-card-jest" hasBottomPadding />,
    );

    const el = container.querySelector('.s-card-jest');

    expect(el).toHaveStyle({ marginBottom: '16px' });
    expect(container).toMatchSnapshot();
  });

  it('should support props extends Card Props', () => {
    const { container } = render(
      <SCard
        className="s-card-jest"
        title="test-title"
        extra={<div>test-extra</div>}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const { container } = render(
      <SCard>
        <div>test-children</div>
      </SCard>,
    );

    expect(container).toMatchSnapshot();
  });
});
