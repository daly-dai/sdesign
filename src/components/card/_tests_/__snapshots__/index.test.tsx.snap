// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SCard > should render children correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    >
      <div>
        test-children
      </div>
    </div>
  </div>
</div>
`;

exports[`SCard > should render correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard > should support custom className 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered test-class css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard > should support custom style 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px; background-color: red;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard > should support hasBottomPadding props 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered s-card-jest css-dev-only-do-not-override-3rel02"
    style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard > should support props extends Card Props 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered s-card-jest css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          test-title
        </div>
        <div
          class="ant-card-extra"
        >
          <div>
            test-extra
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard > should support props style 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-3rel02"
    style="padding: 16px 24px; border-radius: 8px; background-color: red;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard should render children correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1r287do"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    >
      <div>
        test-children
      </div>
    </div>
  </div>
</div>
`;

exports[`SCard should render correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1r287do"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard should support custom className 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered test-class css-dev-only-do-not-override-1r287do"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard should support custom style 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1r287do"
    style="padding: 16px 24px; border-radius: 8px; background-color: red;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard should support hasBottomPadding props 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered s-card-jest css-dev-only-do-not-override-1r287do"
    style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;

exports[`SCard should support props extends Card Props 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered s-card-jest css-dev-only-do-not-override-1r287do"
    style="padding: 16px 24px; border-radius: 8px;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          test-title
        </div>
        <div
          class="ant-card-extra"
        >
          <div>
            test-extra
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
      style="padding: 0px;"
    />
  </div>
</div>
`;
